plugins/modules/opkg.py

```diff
@@ -89,112 +89,82 @@
-    """ Uninstalls one or more packages if installed. """
-
-    p = module.params
-    force = p["force"]
-    if force:
-        force = "--force-%s" % force
-
-    remove_c = 0
-    # Using a for loop in case of error, we can report the package that failed
-    for package in packages:
-        # Query the package first, to see if we even need to remove
-        if not query_package(module, opkg_path, package):
-            continue
-
-        rc, out, err = module.run_command("%s remove %s %s" % (opkg_path, force, package))
-
-        if query_package(module, opkg_path, package):
-            module.fail_json(msg="failed to remove %s: %s" % (package, out))
-
-        remove_c += 1
-
-    if remove_c > 0:
-
-        module.exit_json(changed=True, msg="removed %s package(s)" % remove_c)
-
-    module.exit_json(changed=False, msg="package(s) already absent")
-
-
-def install_packages(module, opkg_path, packages):
-    """ Installs one or more packages if not already installed. """
-
-    p = module.params
-    force = p["force"]
-    if force:
-        force = "--force-%s" % force
-
-    install_c = 0
-
-    for package in packages:
-        if query_package(module, opkg_path, package) and (force != '--force-reinstall'):
-            continue
-
-        rc, out, err = module.run_command("%s install %s %s" % (opkg_path, force, package))
-
-        if not query_package(module, opkg_path, package):
-            module.fail_json(msg="failed to install %s: %s" % (package, out))
-
-        install_c += 1
-
-    if install_c > 0:
-        module.exit_json(changed=True, msg="installed %s package(s)" % (install_c))
-
-    module.exit_json(changed=False, msg="package(s) already present")
-
-
-def main():
-    module = AnsibleModule(
+class Opkg(StateModuleHelper):
+    module = dict(
         argument_spec=dict(
             name=dict(aliases=["pkg"], required=True, type="list", elements="str"),
             state=dict(default="present", choices=["present", "installed", "absent", "removed"]),
             force=dict(default="", choices=["", "depends", "maintainer", "reinstall", "overwrite", "downgrade", "space", "postinstall", "remove",
                                             "checksum", "removal-of-dependent-packages"]),
             update_cache=dict(default=False, type='bool'),
-        )
+        ),
     )
 
-    opkg_path = module.get_bin_path('opkg', True, ['/bin'])
+    def __init_module__(self):
+        self.vars.set("install_c", 0, output=False, change=True)
+        self.vars.set("remove_c", 0, output=False, change=True)
 
-    p = module.params
+        state_map = dict(
+            query="list-installed",
+            present="install",
+            installed="install",
+            absent="remove",
+            removed="remove",
+        )
+
+        def _force(value):
+            if value == "":
+                value = None
+            return cmd_runner_fmt.as_optval("--force-")(value)
```

Comment:
```suggestion
            return cmd_runner_fmt.as_optval("--force-", ignore_none=True)(value, ctx_ignore_None=False)
```

Reply:
None value will be ignored by default, no need to do that.

Reply:
Unfortunately not, because `ignore_none` is initialized to `None` (it should probably be initialized to `True` instead):
https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/cmd_runner.py#L91
If `ignore_none` is not set, the `cmd_runner_fmt.as_optval` command will fail here
https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/cmd_runner.py#L97
Because `ctx_ignore_none` has no default value:
https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/cmd_runner.py#L96

Reply:
The place where the default is enforced is https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/cmd_runner.py#L208

And now that I read all this, I can't help thinking that this is more confusing than it ought to be. I will rethink that a bit and possibly place a PR for CmdRunner later on.

Reply:
> The place where the default is enforced is (...)

I think the assignment `ctx_ignore_value_none=...` is missing there.

Reply:
Indeed. Fixed that already.